### DISCUSSION TITLE
mp3 mit Namen auswählen und einmal abspielen

### DISCUSSION THREAD

**Discussion by FrankvanKessel1** [pschatzmann/arduino-audio-tools#1790] [Q&A] [answered]
Guten Abend.

Ich möchte gerne eine beliebige Mp3 von meiner sd-Karte abspielen. Ich habe ein IA-Thinker Board Audio Kit 2.2 A404.  Und zwar zwischen den vorhandenen mp3 Dateien frei im Code (mit deren Namen) wählen können. Ich möchte aber, dass nur dieses gewählte mp3 Datei einmal gespielt wird und dann ruhe ist.

Hat jemand eine Codbeispiel für mich?

Ich würde später wenn die freie Mp3 (mit Namen) Wahl im Code funktioniert diese per UDP Befehlen wählen und einmal abspielen.

Vielleicht hat ja jemand einen einfachen Code wo ich die Wahl zunächst durchführen/prüfen und verstehen lerne.

per UDP bekomme ich dann später selber hin.

vielen lieben Dank

**Comment by pschatzmann** [comment, marked as answer]
I think you have two options:

- Use the scenario where you play a single mp3 file
- Use the AudioPlayer

I don't have enough info to make a recommendation, but If you always know the file name and path to play I would go for option 1 and if you just want to automaticaly play all (or a directory of) files in sequence, I would go for option 2 

The examples are logically named, so you should be able to find them yourself
e.g [streams-sd_mp3-i2s](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-stream/streams-sd_mp3-i2s) or [player-sd-i2s](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-player/player-sd-i2s)

**Comment by FrankvanKessel1** [comment]
die Variante 1 ist wohl das was ich suche. 
Würde dieser Code, so wie er ist auch auf meinem  IA-Thinker Board Audio Kit 2.2 A404 laufen? 
Sorry wegen meiner Anfängerfragen.

**Reply by FrankvanKessel1** [reply]
ich meinte damit 

[streams-sd_mp3-i2s](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-stream/streams-sd_mp3-i2s)

**Reply by pschatzmann** [reply]
Just replace the output class and make sure that the dip switches are in the right position.

**Reply by FrankvanKessel1** [reply]
vielen vielen dank. läuft...

**Comment by SaKiEQ** [comment]
Morning, I was wondering the same. 

I have an environment that already uses the Player class, to play entire folders of .mp3 or to stream from an URL. 
Now every once in a while I would love to play like a 'boot chime' when the device starts, or and error message as mp3. 

It would be very convenient if the Player class would allow to play a single file  player.setPath(audioFile); ! 
I tried to set setAutoNext(bool next) -> false, but then it seems the whole stops or at least I can not seem to restart it. 

Is there a way to use Player for a single file and me just not figuring it out ?
